organization/src/main/scala/com/improving/app/organization/OrganizationServiceImpl.scala

```diff
@@ -0,0 +1,78 @@
+package com.improving.app.organization
+
+import akka.actor.typed.ActorSystem
+import akka.cluster.sharding.typed.scaladsl.{ClusterSharding, Entity}
+import akka.cluster.typed.{Cluster, Join}
+import akka.grpc.GrpcServiceException
+import akka.persistence.typed.PersistenceId
+import akka.util.Timeout
+import com.google.rpc.Code
+import com.google.rpc.error_details.LocalizedMessage
+import com.improving.app.organization.api.OrganizationService
+import com.improving.app.organization.domain.{ActivateOrganization, EstablishOrganization, GetOrganizationInfo, Organization, OrganizationActivated, OrganizationEstablished, OrganizationInfo, OrganizationSuspended, OrganizationTerminated, SuspendOrganization, TerminateOrganization}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+class OrganizationServiceImpl(sys: ActorSystem[_]) extends OrganizationService {
+  private implicit val system: ActorSystem[_] = sys
+  implicit val timeout: Timeout = 5.minute
+  implicit val executor = system.executionContext
+
+  val sharding = ClusterSharding(system)
+
+  sharding.init(Entity(Organization.TypeKey)(
+    createBehavior = entityContext =>
+      Organization(
+        PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId)
+      )
+  ))
+
+  Cluster(system).manager ! Join(Cluster(system).selfMember.address)
+
+  private def exceptionHandler(exception: Throwable): GrpcServiceException = {
+    GrpcServiceException(
+      code = Code.INVALID_ARGUMENT,
+      message = exception.getMessage,
+      details = Seq(new LocalizedMessage("EN", exception.getMessage))
+    )
+  }
+
+  override def establishOrganization(in: EstablishOrganization): Future[OrganizationEstablished] = {
+    val result = sharding.entityRefFor(Organization.TypeKey, in.organizationId.get.id)
+      .ask(ref => Organization.OrganizationRequestEnvelope(in, ref))
+    result.transform(
+      result => result.getValue.asMessage.sealedValue.organizationEstablished.get,
```

Comment:
I think `result.getValue.asMessage.sealedValue.organizationEstablished.get` can be shortened to `result.getValue.asMessage.getOrganizationEstablishedValue`

Reply:
likewise for the rest of the code changes with `sealedValue`

Reply:
Seems sealedValue doesn't work that way in scalapb, only optional fields on messages.